src/ophyd_async/sim/_blob_detector_writer.py

```diff
@@ -85,17 +85,9 @@ async def collect_stream_docs(
         self, name: str, indices_written: int
     ) -> AsyncIterator[StreamAsset]:
         # When we have written something to the file
-        if indices_written:
-            # Only emit stream resource the first time we see frames in
-            # the file
-            if not self.composer:
-                if not self.path:
-                    raise RuntimeError(f"open() not called on {self}")
-                self.composer = HDFDocumentComposer(self.path, self.datasets)
-                for doc in self.composer.stream_resources():
-                    yield "stream_resource", doc
-            for doc in self.composer.stream_data(indices_written):
-                yield "stream_datum", doc
+        if self.composer:
```

Comment:
I think we should raise if `self.composer is None`
```suggestion
        if self.composer is None:
            msg = "open() not called on {self}"
            raise RuntimeError(msg)
```